### PR TITLE
desktop notifications wallet updater

### PR DESCRIPTION
**Pull request by yhaspel**

### CODE REVIEW

File: app/infra/notificationsService/notificationsService.js [hunk 1]
```diff
@@ -18,6 +20,20 @@ class NotificationsService {
       };
     }
   };
+
+  static checkAppVisibility() {
+    ipcRenderer.send(ipcConsts.CHECK_APP_VISIBLITY);
+    return new Promise<string, Error>((resolve: Function) => {
```
Comment (IlyaVi) [inline app/infra/notificationsService/notificationsService.js]: Response is binary: true or false. Why there are two separate channels for processing the response?

File: app/infra/walletUpdateService/walletUpdateService.js [hunk 2]
```diff
@@ -24,6 +31,10 @@ class WalletUpdateService {
   static listenToDownloadUpdate({ onDownloadUpdateCompleted }: { onDownloadUpdateCompleted: () => void }) {
     ipcRenderer.once(ipcConsts.DOWNLOAD_UPDATE_COMPLETED, () => {
       listenerCleanup({ ipcRenderer, channels: [ipcConsts.DOWNLOAD_UPDATE_COMPLETED] });
+      notificationsService.notify({
```
Comment (IlyaVi) [inline app/infra/walletUpdateService/walletUpdateService.js]: What condition causes the update modal to be displayed? When it's the exact time to notify user that there is an update?

File: app/infra/notificationsService/notificationsService.js (hunk 1 above)
Comment (yhaspel) [reply app/infra/notificationsService/notificationsService.js]: Will consolidate.

File: app/infra/walletUpdateService/walletUpdateService.js (hunk 2 above)
Comment (yhaspel) [reply app/infra/walletUpdateService/walletUpdateService.js]: This is the notification for download completed, not that an update is available.

Comment (IlyaVi) [reply app/infra/walletUpdateService/walletUpdateService.js]: My question is not "what is this" but when "got update" modal is displayed and if it is consistent with notification and it's content

Comment (yhaspel) [reply app/infra/walletUpdateService/walletUpdateService.js]: the content of the notification is "App update downloaded." when the download completes. It is consistent.

File: desktop/main.dev.js [hunk 3]
```diff
@@ -113,6 +113,8 @@ app.on('ready', async () => {
     mainWindow.webContents.send(ipcConsts.REQUEST_CLOSE);
   });
 
+  ipcMain.on(ipcConsts.CHECK_APP_VISIBLITY, () => mainWindow.webContents.send(ipcConsts.IS_APP_VISIBLE, { isAppVisible: mainWindow.isVisible() && mainWindow.isFocused() }));
```
Comment (IlyaVi) [inline desktop/main.dev.js]: Sending object in response is meaningless in this case cause you do destruction and returns destructed value in response handler

**Pull request merged**